File: cdp-agentkit-core/typescript/src/action_providers/twitter/schemas.ts
import { z } from "zod";

/**
 * Input schema for retrieving account details.
 */
export const AccountDetailsSchema = z
  .object({})
  .strip()
  .describe("Input schema for retrieving account details");

/**
 * Input schema for retrieving account mentions.
 */
export const AccountMentionsSchema = z
  .object({
    userId: z
      .string()
      .min(1, "Account ID is required.")
      .describe("The Twitter (X) user id to return mentions for"),
  })
  .strip()
  .describe("Input schema for retrieving account mentions");

/**
 * Input schema for posting a tweet.
 */
export const PostTweetSchema = z
  .object({
    tweet: z.string().max(280, "Tweet must be a maximum of 280 characters."),
  })
  .strip()
  .describe("Input schema for posting a tweet");

/**
 * Input schema for posting a tweet reply.
 */
export const PostTweetReplySchema = z
  .object({
    tweetId: z.string().describe("The id of the tweet to reply to"),
    tweetReply: z
      .string()
      .max(280, "The reply to the tweet which must be a maximum of 280 characters."),
  })
  .strip()
  .describe("Input schema for posting a tweet reply");

File: python/coinbase-agentkit/coinbase_agentkit/action_providers/twitter/schemas.py
"""Schema definitions for Twitter action provider."""
from pydantic import BaseModel, Field


class AccountDetailsInput(BaseModel):
    """Input argument schema for Twitter account details action."""
    pass


class AccountMentionsInput(BaseModel):
    """Input argument schema for Twitter account mentions action."""
    user_id: str = Field(..., description="The Twitter user ID to fetch mentions for")


class PostTweetInput(BaseModel):
    """Input argument schema for posting a tweet."""
    tweet: str = Field(..., description="The text content of the tweet (max 280 characters)")


class PostTweetReplyInput(BaseModel):
    """Input argument schema for posting a tweet reply."""
    tweet_reply: str = Field(..., description="The text content of the reply tweet (max 280 characters)")
    tweet_id: str = Field(..., description="The ID of the tweet to reply to")
